### PR TITLE
Fix: filter managedfields update events

### PR DESCRIPTION
**Pull request by Somefive**
Signed-off-by: Yin Da <yd219913@alibaba-inc.com>

### Description of your changes

Filter Application and ResourceTracker update event by ignoring managedFields changes.
For old k8s version like 1.18.5, the managedField could always update and cause infinite loop, this function helps filter those events and prevent infinite loop.

I have:

- [x] Read and followed KubeVela's [contribution process](https://github.com/oam-dev/kubevela/blob/master/contribute/create-pull-request.md).
- [x] [Related Docs](https://github.com/oam-dev/kubevela.io) updated properly. In a new feature or configuration option, an update to the documentation is necessary. 
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [x] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### CODE REVIEW

**Review by FogDong — APPROVED** [review]

File: pkg/controller/core.oam.dev/v1alpha2/application/application_controller.go [hunk 1]
```diff
@@ -507,6 +506,19 @@ func updateObservedGeneration(app *v1beta1.Application) {
...
 
+// filterManagedFieldChangesUpdate filter resourceTracker update event by ignoring managedFields changes
+// For old k8s version like 1.18.5, the managedField could always update and cause infinite loop
+// this function helps filter those events and prevent infinite loop
+func filterManagedFieldChangesUpdate(e ctrlEvent.UpdateEvent) bool {
+	new, isNewRT := e.ObjectNew.DeepCopyObject().(*v1beta1.ResourceTracker)
+	old, isOldRT := e.ObjectOld.DeepCopyObject().(*v1beta1.ResourceTracker)
+	if !isNewRT || !isOldRT {
+		return true
+	}
```
Comment (wonderflow) [inline pkg/controller/core.oam.dev/v1alpha2/application/application_controller.go:517]: seems these are duplicate code？

File: pkg/controller/core.oam.dev/v1alpha2/application/application_controller.go [hunk 2]
```diff
@@ -444,14 +444,14 @@ func (r *Reconciler) SetupWithManager(mgr ctrl.Manager) error {
...
-				old, ok := e.ObjectOld.DeepCopyObject().(*v1beta1.Application)
-				if !ok {
-					return true
+				new, isNewApp := e.ObjectNew.DeepCopyObject().(*v1beta1.Application)
+				old, isOldApp := e.ObjectOld.DeepCopyObject().(*v1beta1.Application)
+				if !isNewApp || !isOldApp {
+					return filterManagedFieldChangesUpdate(e)
 				}
+				// filter managedFields changes
+				new.ManagedFields = old.ManagedFields
```
Comment (wonderflow) [inline pkg/controller/core.oam.dev/v1alpha2/application/application_controller.go:453]: managedFields could change when generation not changed?

Comment (Somefive) [reply pkg/controller/core.oam.dev/v1alpha2/application/application_controller.go:453]: Yes. k8s 1.18.5 will do that.

File: pkg/controller/core.oam.dev/v1alpha2/application/application_controller.go (hunk 1 above)
Comment (Somefive) [reply pkg/controller/core.oam.dev/v1alpha2/application/application_controller.go:517]: Not exactly. One is to check type Application, here is to check type RT.

**Review by wonderflow — APPROVED** [review]

**Pull request merged**